component/common/loki/client/client_writeto.go

```diff
@@ -18,16 +18,9 @@ import (
 // clientWriteTo implements a wal.WriteTo that re-builds entries with the stored series, and the received entries. After,
 // sends each to the provided Client channel.
 type clientWriteTo struct {
-	series     map[chunks.HeadSeriesRef]model.LabelSet
-	seriesLock sync.RWMutex
-
-	// seriesSegment keeps track of the last segment in which the series pointed by each key in this map was seen. Keeping
-	// this in a separate map avoids unnecessary contention.
-	//
-	// Even though it doesn't present a difference right now according to benchmarks, it will help when we introduce other
-	// calls from the wal.Watcher to the wal.WriteTo like `UpdateSeriesSegment`.
-	seriesSegment     map[chunks.HeadSeriesRef]int
-	seriesSegmentLock sync.RWMutex
+	series        map[chunks.HeadSeriesRef]model.LabelSet
```

Comment:
drive-by: Cleaning up the `seriesCache` locking to use a single lock, since we are not gonna implement `UpdateSeriesSegment` for the time being.